python/ray/experimental/channel/nccl_group.py

```diff
@@ -156,14 +215,20 @@ def send(self, value: "torch.Tensor", peer_rank: int) -> None:
         """
         if self._closed:
             raise RayChannelError("NCCL group has been destroyed.")
+
+        if isinstance(value, GPUFuture):
+            self._send_stream.synchronize()
```

Comment:
Why do we need to synchronize the stream?

Reply:
We observed the following issue: if compute runs on GPU, all recv/compute/send run on GPU. Since the CPU execution loop is not synchronized on GPU, if loops very fast and launches a lot of operations on GPU, which led to issues.

This is a workaround to allow at most 1 send operation in the send stream. I will at least add a comment if no better solution is found at this point.

Reply:
Added a comment.